### ISSUE TITLE
Wrong types of lib was linked when trying to use find_package with third-party scripts

### ISSUE THREAD

**Issue by DWVoid** [microsoft/vcpkg#3060]
Vcpkg package management program version 0.0.104-2018-02-13-50a545fcff4b3c922aa3ef5d0646dc9850f1bb86
Since my project is a cross-platform one, it carries some find_XXX scripts which find modules recursively in given directories with it. When I'm trying to build my cmake project with vcpkg toolchain files [VS2017, VC v141, x64-Release], these scripts reports the DEBUG VERSION OF LIBRARIES INTSEAD OF RELEASE ONES, causing the linkage to fail. 
I wonder if there is any good solutions for this.

**Comment by Ghabry** [comment]
When the debug libs you link in have a suffix like "d" you can modify your find scripts to detect both debug and release version.
If both debug and release have the same filename and you use the Visual Studio generator of CMake you must directly specify -DCMAKE_RELEASE_TYPE=Release. Then the debug build will use the wrong libraries but at least Release should work. I already asked about this issue and that's a CMake limitation.

**Comment by ras0219-msft** [comment]
Just to add to @Ghabry's answer: you can generate two separate VS solutions (one with `-DCMAKE_BUILD_TYPE=Release` and one with `-DCMAKE_BUILD_TYPE=Debug`) which should each work with their respective configurations.

Additionally, VS2017's built-in CMake support should work correctly without any workarounds [1][2].

[1] https://blogs.msdn.microsoft.com/vcblog/2016/11/16/cmake-support-in-visual-studio-the-visual-studio-2017-rc-update/
[2] https://github.com/microsoft/vcpkg/blob/369c60c0f95c351b3be61c16fe9ff6c6fbc807af/docs/EXAMPLES.md#option-b-cmake-toolchain-file

**Comment by DREAMWORLDVOID** [comment]
@ras0219-msft Well, I've tested the VS2017 integration and it simply does not work. I've been looking into the reasons this week and I believe that I found out why. 
In the 92th line in the following file: vcpkg/scripts/buildsystems/vcpkg.cmake, It tells the build type based only on CMAKE_BUILD_TYPE(l have no idea what is the regex for), but the VS integration simply leaves this empty. Instead, it sets the value of CMAKE_CONFIGURATION_TYPES to the desired value, which normally works too. The Debug configuration works because the fallback assumes that it is building in debug mode, while this makes the release configuration fail. 
Currently I am using a workaround to specify the build Type variable directly to make things work, but I do hope that a official solution will be offered.
